internal/dataplane/parser/translate_httproute.go

```diff
@@ -562,3 +591,133 @@ func httpBackendRefsToBackendRefs(httpBackendRef []gatewayv1beta1.HTTPBackendRef
 	}
 	return backendRefs
 }
+
+// ---------------------
+// POC methods
+// ---------------------
+
+func splitHTTPRouteRules(httpRoutes []*gatewayv1beta1.HTTPRoute) []gatewayv1beta1.HTTPRoute {
+	newHTTPRoutes := []gatewayv1beta1.HTTPRoute{}
+	hostnamedRoutes := []gatewayv1beta1.HTTPRoute{}
+	for _, route := range httpRoutes {
+		if len(route.Spec.Hostnames) == 0 {
+			hostnamedRoutes = append(hostnamedRoutes, *route)
+			continue
+		}
+		for _, hostname := range route.Spec.Hostnames {
+			hostnamedRoute := route.DeepCopy()
+			hostnamedRoute.Spec.Hostnames = []gatewayv1beta1.Hostname{hostname}
+			hostnamedRoutes = append(hostnamedRoutes, *hostnamedRoute)
+		}
+	}
+	for _, route := range hostnamedRoutes {
+		for _, rule := range route.Spec.Rules {
+			for _, match := range rule.Matches {
+				newRoute := route.DeepCopy()
+				newRoute.Spec.Rules = []gatewayv1beta1.HTTPRouteRule{
+					{
+						Matches: []gatewayv1beta1.HTTPRouteMatch{
+							match,
+						},
+						BackendRefs: rule.BackendRefs,
+						Filters:     rule.Filters,
+					},
+				}
+				newHTTPRoutes = append(newHTTPRoutes, *newRoute)
+			}
+		}
+	}
+
+	return newHTTPRoutes
+}
+
+func sortHTTPRoutes(routes []gatewayv1beta1.HTTPRoute) {
+	sort.SliceStable(routes, func(i, j int) bool {
+		hasHostnamei := hasHostnames(routes[i])
+		hasHostnamej := hasHostnames(routes[j])
+
+		matchi := routes[i].Spec.Rules[0].Matches[0]
+		matchj := routes[j].Spec.Rules[0].Matches[0]
+		pathMatchingi := matchi.Path
+		pathMatchingj := matchj.Path
+		isExactPathMatchingi := isPathMatchingExact(pathMatchingi)
+		isExactPathMatchingj := isPathMatchingExact(pathMatchingj)
+		isPrefixPathMatchingi := isPathMatchingPrefix(pathMatchingi)
+		isPrefixPathMatchingj := isPathMatchingPrefix(pathMatchingj)
+
+		namespacedNamei := fmt.Sprintf("%s/%s", routes[i].Namespace, routes[i].Name)
+		namespacedNamej := fmt.Sprintf("%s/%s", routes[j].Namespace, routes[j].Name)
+
+		switch {
+		// hostname matching
+		case hasHostnamei && !hasHostnamej:
+			return true
+		case !hasHostnamei && hasHostnamej:
+			return false
+		case hasHostnamei && hasHostnamej && len(routes[i].Spec.Hostnames[0]) != len(routes[j].Spec.Hostnames[0]):
```

Comment:
The specification did not consider how to compare hostnames when there are multiple hostnames in one `HTTPRoute`?